README.md

```diff
@@ -16,72 +20,199 @@ make
-instructions for Operator Courier to obtain an account token. 
+## Deploy to OpenShift 4.7+ using OLM
+
+To install this operator on OpenShift 4 for end-to-end testing, make sure you have access to a quay.io (https://quay.io/) account to create
+an application repository. Follow the [authentication](https://github.com/operator-framework/operator-courier/#authentication)
+instructions for Operator Courier to obtain an account token.
 This token is in the form of "basic XXXXXXXXX" and both words are required for the command.
 
-If pushing to another quay repository, replace _kiegroup_ with your username or other namespace. 
-Also note that the push command does not overwrite an existing repository, 
-and it needs to be deleted before a new version can be built and uploaded. 
-Once the bundle has been uploaded, create an [Operator Source](https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md#linking-the-quay-application-repository-to-your-openshift-40-cluster) 
+Also note that the push command does not overwrite an existing repository,
+and it needs to be deleted before a new version can be built and uploaded.
+Once the bundle has been uploaded, create an [Operator Source](https://github.com/k8s-operatorhub/community-operators/)
 to load your operator bundle in OpenShift.
 
-**Create your own index image**
 
-Requires [opm](https://github.com/operator-framework/operator-registry/releases) v1.15.3+ -
+**To create the bundle image follow the steps**
 
-```bash
-USERNAME=tchughesiv
-VERSION=$(go run getversion.go -csv)
-IMAGE=quay.io/${USERNAME}/rhpam-operator-bundle
-BUNDLE=${IMAGE}:${VERSION}
+- Create your own bundle
+- Push the bundle on the container registry
+- Build the index
+- Push the index on the container registry
+- Disable default catalog sources on Openshift
+- Write your Catalog-source
+- Create your catalog source on Openshift
+- Write your Subscription
+- Create your Subscription on Openshift
+
+**To Restore your cluster from your bundle image changes follow the steps**
+
+- Cleanup your catalog-source
+
+### Create your own Bundle
+
+i.e. 7.12.0-1 version
+Remove the following line from deploy/olm-catalog/dev/7.12.0-1/manifest/businessautomation-operator.clusterserviceversion.yaml
+
+```console
+replaces: businessautomation-operator.<last-version>
+```
+Set your registry id, like quay username 
+with USERNAME as env
+
+```console
+export USERNAME=<your-registry-id>
+```
+
+activate Cekit and run the following command
+
+
+```console
 $ make bundle-dev
+```
+
+the last log line is something like this:
+```console
+INFO  Image built and available under following tags: quay.io/<your_quay_username>/rhpam-operator-bundle:7.12.1, quay.io/${USERNAME}/rhpam-operator-bundle:latest
+```
+###  Push the bundle on the container registry
+
+VERSION=$(go run getversion.go)
+
+USERNAME=<your_quay_username>
```

Comment:
Your idea is to reuse the QUAY_USERNAME in the build script instead of USERNAME ?
The use case is that when you enter your credential for the first time a token is saved on the .docker/config.json and then the QUAY_USERNAME is no more needed because you just use the command podman/docker login quay.io without enter credentials.